packages/autocomplete-js/src/autocomplete.ts

```diff
@@ -217,6 +217,7 @@ export function autocomplete<TItem extends BaseItem>(
     return () => {
       if (panelContainerElement.contains(panelElement)) {
         panelContainerElement.removeChild(panelElement);
+        panelContainerElement.classList.remove('aa-Detached');
```

Comment:
This looks like a good solution, although I wonder if we should be using `setIsModalOpen(false)` instead of the entire block 🤔

Reply:
`setIsModalOpen` looks like it is only for detached mode, but this block also handles removing the panel from it's container in non-detached mode